sdk/storage/azblob/container/client.go

```diff
@@ -329,3 +329,40 @@ func (c *Client) GetSASURL(permissions sas.ContainerPermissions, expiry time.Tim
 
 	return endpoint, nil
 }
+
+// NewBatchBuilder creates an instance of BatchBuilder with the specified values.
+//   - cred - an Azure AD credential, typically obtained via the azidentity module
+//   - options - client options; pass nil to accept the default values
+func (c *Client) NewBatchBuilder(cred azcore.TokenCredential, options *ClientOptions) *BatchBuilder {
```

Comment:
I don't think we need ClientOptions for this since it is not really a client

Reply:
Presumably we'd have `BatchBuilderOptions` though yes?

Reply:
We could but I don't think it's necessary. I'm struggling to think of anything that could show up here in the future. The BatchBuilder is really just concatenating items together - it never talks to a service. We'd have to have an option that applies to the entire batch (not individual items).

Reply:
I see.  I suppose we also have an escape hatch; we could add a method to the BB to configure it post-construction if necessary.

Reply:
Agreed.  Then I worry that omitting an options type might pidgeon-hole us in the future.

Reply:
@JeffreyRichter: In case user has a custom user-agent string that is set using these client-options and same is case for custom pipeline policies. If we do not take the client-options we lose this flexibility. For e.g. user-agent will be marked only on the outer request and not at every sub-request level. This means when FE iterates the batch and executes individual sub-request user-agent will not be present and this might skew some of the metrics collected for billing and analysis purpose (not sure each individual sub-request is billed separately or not, but assuming that will be the case).

Reply:
The BatchBuilder's pipeline is not for sending a request to a service; it is for building an in-memory buffer composed of several batch items that ultimately SubmitBatch will send to a service There is no such concept of a user-agent string for the BatchBuilder. For the same reason, there is no need for customer customization of the BatchBuilder's pipeline. 

The purpose of the UA string is for the service to know what SDK is making the request; it doesn't make sense for each item to be generated by different SDKs. The customer is billed for each SubmitBatch operation; they are NOT billed for each item in the batch. But, this also has nothing to do with user-agent headers. 

As a side note: A service can NEVER trust data provided by a customer. If the UA string were used for billing, then customers would set this to whatever they want for their advantage - services can never be designed to trust this.